jni/src/test/java/com/horizen/certnative/NaiveThresholdSignatureWKeyRotationProofTest.java

```diff
@@ -149,9 +119,199 @@ private void testCreateVerifyRandomProof(int numCustomFields, String snarkPkPath
             updatedMasterKeysSkSignaturesList,
             updatedMasterKeysMkSignaturesList
         );
+    }
+
+    private List<SchnorrKeyPair> setupKeyPairList(int keyCount) {
+        List<SchnorrKeyPair> keyPairList = new ArrayList<>();
+        for (int i = 0; i<keyCount; i++) {
+            SchnorrKeyPair keyPair = SchnorrKeyPair.generate();
+
+            assertNotNull("Key pair generation was unsuccessful.", keyPair);
+            assertTrue("Public key verification failed.", keyPair.getPublicKey().verifyKey());
+
+            keyPairList.add(keyPair);
+        }
+        return keyPairList;
+    }
+
+    private List<BackwardTransfer> getBackwardTransferList(int count) {
+        Random r = new Random();
+        List<BackwardTransfer> btList = new ArrayList<>();
+        for(int i = 0; i < count; i++)
+            btList.add(BackwardTransfer.getRandom(r));
+        return btList;
+    }
+
+    private List<SchnorrSignature> createSignatures(
+        WithdrawalCertificate aWithdrawalCertificate,
+        List<SchnorrKeyPair> keyPairList
+    ) {
+        List<SchnorrSignature> signatureList = new ArrayList<>();
+        FieldElement msgToSign = NaiveThresholdSignatureWKeyRotation.createMsgToSign(aWithdrawalCertificate);
+        for (int i = 0; i<keyCount; i++) {
+            if (i < threshold) {
+                signatureList.add(keyPairList.get(i).signMessage(msgToSign));
+            } else {
+                signatureList.add(new SchnorrSignature());
+            }
+        }
+        return signatureList;
+    }
+
+    @Test
+    public void testSigningKeyRotation() throws Exception {
```

Comment:
I would add a flag `usePrevCert` as in the `testCreateVerifyRandomProof` so that you can run the test both with and without a previous certificate